### DISCUSSION TITLE
no/unstable connection with HMS-1600-4T and HMS-2000-4T with 2 different boards

### DISCUSSION THREAD

**Discussion by miknuc** [tbnobody/OpenDTU#1491] [Q&A]
Hi 
I am currently setting up some new panels, with 2 HMS-1660-4T and 1 HMS-2000-4T. I learned that I need a CMT2300A. I bought this board from 
https://wechselrichter-versand.de/openDTU-S-fuer-Hoymiles-Serie-HMT-HMS-Bausatz/HY10012

I started connecting it to my HMS-2000-4T.
I build the image for the esp32 using VSCode based on the current version in the github repositoy.

After setting up the board and configuring the I did not get any connection. I think the connection to the CMT2003A as I see this entry in the log file after a restart:
```
14:58:54.670 > Initialize Hoymiles interface... CMT: Connection successful
14:58:54.805 >   Setting CMT target frequency... 
14:58:54.810 >   Setting radio PA level... 
14:58:54.810 > CMT TX power set to 0 dBm
```
The info dialog also displays 

CMT2300A Status | konfiguriert
-CMT2300A Chip-Status verbunden

However I only get these no connection errors:
`14:58:58.932 > Fetch inverter: 116480151994
14:59:03.953 > Fetch inverter: 116484664257
14:59:03.953 > TX RealTimeRunData 865.00 MHz --> 15 84 66 42 57 80 15 62 56 80 0B 00 65 4F 88 A7 00 00 00 00 00 00 00 00 A6 10 7B 
14:59:04.500 > RX Period End
14:59:04.500 > All missing
14:59:04.500 > Nothing received, resend whole request
14:59:04.505 > TX RealTimeRunData 865.00 MHz --> 15 84 66 42 57 80 15 62 56 80 0B 00 65 4F 88 A7 00 00 00 00 00 00 00 00 A6 10 7B 
14:59:05.055 > RX Period End
14:59:05.055 > All missing
14:59:05.055 > Nothing received, resend whole request
14:59:05.061 > TX RealTimeRunData 865.00 MHz --> 15 84 66 42 57 80 15 62 56 80 0B 00 65 4F 88 A7 00 00 00 00 00 00 00 00 A6 10 7B 
14:59:05.610 > RX Period End
14:59:05.611 > All missing
14:59:05.611 > Nothing received, resend whole request
14:59:05.616 > TX RealTimeRunData 865.00 MHz --> 15 84 66 42 57 80 15 62 56 80 0B 00 65 4F 88 A7 00 00 00 00 00 00 00 00 A6 10 7B 
14:59:06.167 > RX Period End
14:59:06.167 > All missing
14:59:06.167 > Nothing received, resend whole request
14:59:06.172 > TX RealTimeRunData 865.00 MHz --> 15 84 66 42 57 80 15 62 56 80 0B 00 65 4F 88 A7 00 00 00 00 00 00 00 00 A6 10 7B`

I tried to increase the transmit power up to the maximum, without success.
The board is powered with a good charger (for apple Ipad), the distance between the board was ~ 20 cm. I also removed the antenna from the board, without success.

To reduce potential errors I
- bought another board with a CMT2003A  (https://www.ebay.de/itm/OpenDTU-Platine-ESP32-DIY-PCB-Balkonkraftwerk-Hoymiles-HM-HMS-HMT-Ahoy-Open-DTU-/295768693772?var=)
- configured one of my HMT-1600

I was successful to setup a connection once and read data from the inverter (good match with the data from the shelly that is monitoring the inverter). I did this test without an external antenna and was only able to see the inverter near to the board. In order to get also data from the HMS-2000 I increased the transmit power to maximum. Since that moment I did not get any connection again.

Since I was able to get this single connection with one of my inverter, I do assume that my hardware setup works. Since these boards do not require any cabeling and I used different power supplies I think its not the hardware. 

And as I got one connection done I doubt that I have a typo in the IDs of the inverter.

I have an old homematic network that also uses 868Mhz. I did not see any complaints on homematic side about transmission errors but I am wondering whether this could be an influence? 

Does anyone have an idea?

**Comment by ms49434** [comment]
Hi Miknuc,

Did you install the capacitors on the board as suggested in the installation instructions?
Did you try to power the unit with a power supply instead of a charger?
Did you try to set the initial RX/TX frequency of the CMT2300A in OpenDTU to 868 MHz or better 868.75 MHz?
Did you try to lower transmit power?

**Reply by miknuc** [reply]
Hi 
Thank you for the quick answer!
See my replies:
1. yes I added the 100uF capacitors
2. yes, with the first board. The second board has no pins for a direct 5V supply
3. I will try this tomorrow
4. no not below 0 db. why could this help?

**Reply by miknuc** [reply]
Good morning from Hamburg

Changing the frequency to 868.75 helped to get data from the HMS-1600. Still no response from the HMS-2000. I also moved the board from direct distance to HMS-1600 to HMS-2000, with no success. Seems to be a problem with the HMS-2000. 
I think I will get a Hoymiles DTU as a next step to ensure that it's not simply a broken inverter (though it's doing the main job of inverting).
I also reduce the transmit power up to -10db. I can still reach the HMS-1600 but still not answer from the HMS-2000.
Thank you for the advice :)

**Comment by LEO83-Smart** [comment]
Hello!

I have the same problem with the board from "wechselrichter-versand.de" The Board works, the display shows the wattages of 0W, i have all done like the installation instructions, but ther is no conection to my Hoymiles HMT-2250 possibel. I turned off an on the Hoymiles, no connection, the board is in a distance of 3 meters to the Hoymiles. The BIN file for HMT/HMS was installed and aktivatet, the Ser.Nr. of the Hoymiles is correct....no connection.
I tried the frequence 868,75, nothing has changed, radio down to -5dB, then to -10dB or up to 5dB, then 10dB........nothing changes, no connection to my HMT-2250.

Some ideas?

Greetings, Peter

**Reply by ms49434** [reply]
Do you see green lights for the CMT2300A status/chip status in Info > System?

**Comment by LEO83-Smart** [comment]
The miraculous healing. Today at 6:57 o clock in the morning, the HMT-2250 begins to communicate. I'm really happy.
It works with 868,00 Mhz and +5dB.

But only one thing. I see only one phase of three, why? The voltage in the one shwon phase is 400 Volt, so there must be three phases at the system working. Is it possible to change settings to have measuring about all phases.

Greetings,

Peter

**Reply by miknuc** [reply]
I also learned that some of these issues seem to be time dependent. I tried yesterday whether my second board would also work with the frequency of 868.75. It did not and when I then started to use my first board again I was not able to get a connection. I switched the board off for an hour and after that it continued to work. I remember to have heard something about a transmitting budget for devices on this frequency.

**Reply by janhieber** [reply]
I also see only one phase. 
I think it's OK, I guess OpenDTU does not yet support reading/displaying the three phase inverters correctly.
